pkg/adaptation/adaptation.go

```diff
@@ -334,12 +334,27 @@ func (r *Adaptation) startPlugins() (retErr error) {
 		}
 
 		if err := p.start(r.name, r.version); err != nil {
-			return err
+			return fmt.Errorf("failed to start NRI Plugin %q: %w", name, err)
 		}
 
 		plugins = append(plugins, p)
 	}
 
+	syncPlugins := func(ctx context.Context, pods []*PodSandbox, containers []*Container) ([]*ContainerUpdate, error) {
+		var updates []*ContainerUpdate
+		for _, plugin := range plugins {
+			us, err := plugin.synchronize(ctx, pods, containers)
+			if err != nil {
+				return nil, fmt.Errorf("failed to sync NRI Plugin %q: %w", plugin.name(), err)
+			}
+			updates = append(updates, us...)
+		}
+		return updates, nil
+	}
+	if err := r.syncFn(noCtx, syncPlugins); err != nil {
```

Comment:
do we need to be in r.lock here..

Reply:
Already locked in `Start`
https://github.com/containerd/nri/blob/fa3ab8fbf3c5e58f652a7a54f750f9b6a02416f7/pkg/adaptation/adaptation.go#L140-L150